src/Storages/System/StorageSystemDetachedParts.cpp

```diff
@@ -16,18 +17,57 @@ StorageSystemDetachedParts::StorageSystemDetachedParts(const StorageID & table_i
 {
     StorageInMemoryMetadata storage_metadata;
     storage_metadata.setColumns(ColumnsDescription{{
-        {"database", std::make_shared<DataTypeString>()},
-        {"table", std::make_shared<DataTypeString>()},
-        {"partition_id", std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
-        {"name", std::make_shared<DataTypeString>()},
-        {"disk", std::make_shared<DataTypeString>()},
-        {"reason", std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"database",         std::make_shared<DataTypeString>()},
+        {"table",            std::make_shared<DataTypeString>()},
+        {"partition_id",     std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
+        {"name",             std::make_shared<DataTypeString>()},
+        {"bytes_on_disk",    std::make_shared<DataTypeUInt64>()},
+        {"disk",             std::make_shared<DataTypeString>()},
+        {"path",             std::make_shared<DataTypeString>()},
+        {"reason",           std::make_shared<DataTypeNullable>(std::make_shared<DataTypeString>())},
         {"min_block_number", std::make_shared<DataTypeNullable>(std::make_shared<DataTypeInt64>())},
         {"max_block_number", std::make_shared<DataTypeNullable>(std::make_shared<DataTypeInt64>())},
-        {"level", std::make_shared<DataTypeNullable>(std::make_shared<DataTypeUInt32>())}
+        {"level",            std::make_shared<DataTypeNullable>(std::make_shared<DataTypeUInt32>())}
     }});
     setInMemoryMetadata(storage_metadata);
 }
+static void calculateTotalSizeOnDiskImpl(const DiskPtr & disk, const String & from, UInt64 & total_size)
+{
+    /// Files or directories of detached part may not exist. Only count the size of existing files.
+    if (disk->isFile(from))
+    {
+        try
+        {
+            total_size += disk->getFileSize(from);
```

Comment:
isFile can also throw if file was concurrently deleted, so put it also in try catch with the getFileSize? and also below `calculateTotalSizeOnDiskImpl` also need to be put near iterateDirectory in try catch because of the same reason. But summing all of this - it looks you just need to put all this calculateTotalSizeOnDiskImpl into try catch, so probably just add try catch in calculateTotalSizeOnDisk when calling calculateTotalSizeOnDiskImpl and remove all try-catches from calculateTotalSizeOnDiskImpl?